### ISSUE TITLE
Typescript doesn't compile when used with Pixi version 5.3.0

### ISSUE THREAD

**Issue by getkey** [michalochman/react-pixi-fiber#185]
### Description

After upgrading to Pixi 5.3.0, I started getting compilation errors:

<details>
<summary>Click to view the full error</summary>

```
ERROR in /mnt/data/devel/ble/src/components/Block.tsx
./src/components/Block.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Block.tsx(28,4)
      TS2322: Type '{ children: (Element | Element[])[]; }' is not assignable to type 'InteractiveComponent'.
  Property 'children' is incompatible with index signature.
    Type '(Element | Element[])[]' is not assignable to type '(event: any) => void'.
      Type '(Element | Element[])[]' provides no match for the signature '(event: any): void'.

ERROR in /mnt/data/devel/ble/src/components/Block.tsx
./src/components/Block.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Block.tsx(29,5)
      TS2322: Type '{ fill: number; points: Point[]; interactive: true; pointerdown: (ev: InteractionEvent) => void; isSelected: boolean; }' is not assignable to type 'InteractiveComponent'.
  Property 'fill' is incompatible with index signature.
    Type 'number' is not assignable to type '((event: any) => void) | undefined'.

ERROR in /mnt/data/devel/ble/src/components/Block.tsx
./src/components/Block.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Block.tsx(37,6)
      TS2322: Type '{ fill: number; x: number; y: number; key: string; radius: number; interactive: true; pointerdown: (ev: InteractionEvent) => void; isSelected: boolean; }' is not assignable to type 'InteractiveComponent'.
  Property 'fill' is incompatible with index signature.
    Type 'number' is not assignable to type '((event: any) => void) | undefined'.

ERROR in /mnt/data/devel/ble/src/components/Grid.tsx
./src/components/Grid.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Grid.tsx(48,4)
      TS2322: Type '{ position: { x: number; y: number; }; texture: RenderTexture; width: number; height: number; interactive: true; pointerdown: (ev: any) => void; }' is not assignable to type 'InteractiveComponent'.
  Property 'position' is incompatible with index signature.
    Type '{ x: number; y: number; }' is not assignable to type '(event: any) => void'.
      Type '{ x: number; y: number; }' provides no match for the signature '(event: any): void'.

ERROR in /mnt/data/devel/ble/src/components/Grid.tsx
./src/components/Grid.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Grid.tsx(54,18)
      TS7006: Parameter 'ev' implicitly has an 'any' type.

ERROR in /mnt/data/devel/ble/src/components/Level.tsx
./src/components/Level.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/Level.tsx(16,4)
      TS2322: Type '{ children: (string | number | boolean | {} | Element | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | ReactNodeArray | ReactPortal | null | undefined)[]; ... 4 more ...; interactive: true; }' is not assignable to type 'InteractiveComponent'.
  Property 'children' is incompatible with index signature.
    Type '(string | number | boolean | {} | Element | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | ReactNodeArray | ReactPortal | null | undefined)[]' is not assignable to type '(event: any) => void'.
      Type '(string | number | boolean | {} | Element | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | ReactNodeArray | ReactPortal | null | undefined)[]' provides no match for the signature '(event: any): void'.

ERROR in /mnt/data/devel/ble/src/components/PixiApp.tsx
./src/components/PixiApp.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/PixiApp.tsx(39,8)
      TS2322: Type '{ x: number; y: number; key: string; isSelected: boolean; interactive: true; pointerdown: (ev: InteractionEvent) => void; rotation: number; }' is not assignable to type 'InteractiveComponent'.
  Property 'x' is incompatible with index signature.
    Type 'number' is not assignable to type '((event: any) => void) | undefined'.

ERROR in /mnt/data/devel/ble/src/components/PixiApp.tsx
./src/components/PixiApp.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/PixiApp.tsx(76,8)
      TS2322: Type '{ x: number; y: number; key: string; isSelected: boolean; interactive: true; pointerdown: (ev: InteractionEvent) => void; rotation: number; }' is not assignable to type 'InteractiveComponent'.
  Property 'x' is incompatible with index signature.
    Type 'number' is not assignable to type '((event: any) => void) | undefined'.

ERROR in /mnt/data/devel/ble/src/components/ProgressiveText.tsx
./src/components/ProgressiveText.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/ProgressiveText.tsx(19,5)
      TS2322: Type '{ style: { fontFamily: string[]; fontWeight: string; fontSize: number; fill: number; align: string; }; children?: ReactNode; }' is not assignable to type 'InteractiveComponent'.
  Property 'style' is incompatible with index signature.
    Type '{ fontFamily: string[]; fontWeight: string; fontSize: number; fill: number; align: string; }' is not assignable to type '(event: any) => void'.
      Type '{ fontFamily: string[]; fontWeight: string; fontSize: number; fill: number; align: string; }' provides no match for the signature '(event: any): void'.

ERROR in /mnt/data/devel/ble/src/components/ProgressiveText.tsx
./src/components/ProgressiveText.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/ProgressiveText.tsx(33,4)
      TS2322: Type '{ font: string; tint: number; children?: ReactNode; }' is not assignable to type 'InteractiveComponent'.
  Property 'font' is incompatible with index signature.
    Type 'string' is not assignable to type '((event: any) => void) | undefined'.
```

</details>

I tried to create a [reduced test case](https://codesandbox.io/s/sharp-easley-nge8u?file=/src/Bunny.tsx) but I was unable to reproduce the issue I encountered in my repo.

### Steps to reproduce

1. clone https://github.com/getkey/ble/
2. `git checkout pixi-v5.3-issue`
3. `yarn dev`

### Additional info

- react-pixi-fiber version: 0.14.0
- React version: 16.13.1
- ReactDOM version: 16.13.1
- PixiJS version: 5.3.0

**Comment by michalochman** [comment]
Thanks for reporting @getkey. Looks like there's a major TS rewrite of Pixi in 5.3.0.
All types in this project are contained to a single [index.d.ts](https://github.com/michalochman/react-pixi-fiber/blob/27efcff1c9c83ca809fc9eb68b632e252e24f1ce/index.d.ts) file, so I'd start looking there.

I'll not be able to have a look at this for the next couple weeks unfortunately.

**Comment by ldd** [comment]
I confirm that downgrading to `Pixi 5.2.4` temporarily fixes the issue.

**Comment by michalochman** [comment]
@ldd @getkey @yuensunn the issue should be fixed in #187, can you confirm it works in your codebase before I merge this? You can also play using the typescript sandbox linked in that PR.

I have checked `pixi.js@5.0.4`, `pixi.js@5.2.4` and `pixi.js@5.3.0` and [test/typescript/index.tsx](https://github.com/michalochman/react-pixi-fiber/blob/27efcff1c9c83ca809fc9eb68b632e252e24f1ce/test/typescript/index.tsx) file compiles in each after my changes agains .

**Comment by ldd** [comment]
tl;dr: fix works

I can confirm that at least on my project, the pull request fix works with `pixi.js@5.3.0`.

What I did:

- yarn install using [these instructions](https://ci.codesandbox.io/status/michalochman/react-pixi-fiber/pr/187/builds/39100):
`yarn add https://pkg.csb.dev/michalochman/react-pixi-fiber/commit/8a9ff332/react-pixi-fiber
`
- upgrade pixi js version:
`yarn add pixi.js`

Then I restarted my project and everything worked correctly.

**Comment by michalochman** [comment]
Released in `react-pixi-fiber@0.14.1`

**Comment by getkey** [comment]
Sorry, I was on the road. 🙂 I can confirm this works too! Thanks!
